src/main/kotlin/org/jitsi/nlj/rtp/codec/vp8/Vp8Packet.kt

```diff
@@ -44,7 +44,8 @@ class Vp8Packet(
     override fun clone(): Vp8Packet {
         val clone = Vp8Packet(buffer.cloneFromPool(), offset, length)
         clone.isKeyframe = isKeyframe
-        clone.trackEncodings = trackEncodings?.copyOf()
+        // TODO can we ask the superclass to clone its own fields?
```

Comment:
i've thought about this same thing because i hate having to copy all the parent fields in every subclass, but it's tricky because we want to have an instance of the subclass type, so we can't just invoke the parent clone (since that gives us an instance of the parent type) so i dunno how we'd do it...we could have a separate helper `cloneFieldsTo` or something? so we'd do like
```
fun clone(): Vp8Packet {
  val clone = Vp8Packet(...)
  super.cloneFieldsTo(clone)
  cloneFIeldsTo(clone)
}

protected fun cloneFieldsTo(other: Vp8Packet) {
// clone vp8 fields
}
```
where `VideoRtpPacket` then looks like:
```
protected fun cloneFieldsTo(other: VideoRtpPacket) {
  other.qualityIndex = qualityIndex
  ...
}

fun clone(): VideoRtpPacket {
  val clone = VideoRtpPacket(...)
  cloneFIeldsTo(clone)
  return clone
}
```
i dunno

Reply:
That's what I had in mind too